### ISSUE TITLE
Various accessibility issues with VoiceOver on macOS

### ISSUE THREAD

**Issue by Isaac-Leonard** [qgis/QGIS#53975]
### What is the bug or the crash?

For context I am a blind university student currently taking a GIS course that requires using Qgis.
While I understand that most of the program can't be made accessibly easily without extensive work there are some smaller issues with simply using the interface that would be extremely useful to have working.
Write now many of the buttons don't seem to click when activated, likely due to not using actual button elements/components.
Tab key navigation is largely broken and often gets fully stuck in a "type to locate" box.
Many of the combo boxes for picking layers or features to look at don't seem to work or if they do they give no feedback that VoiceOver can pick up on.
The browser has issues with not being able to focus on it with VoiceOver and adding layers through menus instead of the browser fails for some reason with no errors or explanations.

I'm happy to help do the work to fix these issues if I can have some guidance on where in the codebase I should look at.
I'm on the latest version of macOS with an m2 Mac and have the latest version of Qgis installed from homebrew.
Let me know if any more information is needed.

### Steps to reproduce the issue

I'd just suggest attempting to use Qgis with VoiceOver enabled and seeing what works and what doesn't.
For the main issues I believe It may be sufficient to just try using it with just the keyboard, no mouse or trackpad, without VoiceOver enabled.

### Versions

3.32.1

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)
